packages/react-components/src/components/Toast.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import cx from 'classnames';
+
+// TODO: remove and use the Icon wrapper with correct icon after migration
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+
+import { ToastIcon } from './ToastIcon';
+
+const baseClass = 'lc-toast';
+
+export enum Variants {
+  Success = 'success',
+  Warning = 'warning',
+  Error = 'error',
+  Info = 'info',
+  Notification = 'notification',
+}
+
+interface IToastActionProps {
+  handler: () => void;
+  label: string;
+  closeOnClick?: boolean;
+}
+
+export interface IToastProps {
+  action: IToastActionProps;
+  className?: string;
+  id: string;
+  removable: boolean;
+  variant: Variants;
+  onClose: () => void;
+}
+
+export const Toast: React.FC<IToastProps> = ({
+  action,
+  className,
+  id,
+  children,
+  removable,
+  variant,
+  onClose,
+}) => {
+  const mergedClassNames = cx(
+    baseClass,
+    {
+      [`${baseClass}--${variant}`]: variant,
+    },
+    className
+  );
+
+  const onActionClick = (action: IToastActionProps) => {
+    if (action && action.closeOnClick && onClose) {
```

Comment:
Wouldn't check for `action` be enough? I mean we defined `IToastActionProps` ...

Reply:
Here, we need to verify if the optional option `closeOnClick` is defined because this will execute the specific effect of clicking the action button (will close or no)